### ISSUE TITLE
Configuration-Cache compatibility

### ISSUE THREAD

**Issue by Vampire** [ph4r05/javacard-gradle-plugin#8]
Also something to consider is the configuration cache.
It is an upcoming Gradle feature that is currently experimental.
It can cache the result of the configuration phase.
Currently this plugin is only partly safe for that.
If one does not use the configuration DSL to configure certain things, the `local.properties` file is inspected as well as environment variables and system properties.
Those accesses need to be explicitly declared for being configuration cache compatible.

You can read more about the configuration cache at https://docs.gradle.org/current/userguide/configuration_cache.html.
As far as I have seen, only those places are cache-unsafe and they can be work-arounded by configuring the properties via DSL, but maybe something to consider for the future or at least put into the readme.

**Comment by ph4r05** [comment]
Thanks for a suggestion. It is good to keep track of this. But due to lack of free time I put this on hold.

**Comment by Vampire** [comment]
Yeah, sure, if you configure the values via DSL there is also no problem in the current version as far as I have seen.
So there is an easy work-around, just something to do in the future to support it natively and thought I add it to your backlog here. :-)

**Comment by Vampire** [comment]
Btw. configuration cache is stable since some versions and soon will even be enabled by default
